### PR TITLE
[Feature] support run StarRocks on mac m1 using docker

### LINKED ISSUE #11826: [Feature] support run StarRocks on mac m1 using docker
## Feature request
Support run be and fe on mac m1 platform. Build dev env docker image for it.

### PR DESCRIPTION
**Pull request by luohaha**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #11826

## Problem Summary(Required) ：
Support run be and fe on mac m1 platform. Build dev env docker image for it. Do some change so we can build docker image and compile be/fe on Mac m1 platform.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by chaoyli — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

File: build.sh [hunk 1]
```diff
@@ -112,6 +112,17 @@ fi
 if [[ -z ${USE_SSE4_2} ]]; then
     USE_SSE4_2=ON
 fi
+# detect cpuinfo
+if [[ -z $(grep -o 'avx[^ ]*' /proc/cpuinfo) ]]; then
+    USE_AVX2=OFF
+    echo "cpu not support AVX"
```
Comment (chaoyli) [inline build.sh:118]: No need to print it.

File: build.sh [hunk 2]
```diff
@@ -112,6 +112,17 @@ fi
...
+fi
+
+if [[ -z $(grep -o 'sse[^ ]*' /proc/cpuinfo) ]]; then
+    USE_SSE4_2=OFF
+    echo "cpu not support SSE"
```
Comment (chaoyli) [inline build.sh:123]: No need to print the echo

File: build.sh (hunk 1 above)
Comment (luohaha) [reply build.sh]: Done

**Comment by luohaha** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by waittttting — APPROVED** [review]

**Review by imay — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

File: build.sh [hunk 3]
```diff
@@ -112,6 +112,15 @@ fi
 if [[ -z ${USE_SSE4_2} ]]; then
     USE_SSE4_2=ON
 fi
+# detect cpuinfo
+if [[ -z $(grep -o 'avx[^ ]*' /proc/cpuinfo) ]]; then
+    USE_AVX2=OFF
+fi
+
+if [[ -z $(grep -o 'sse[^ ]*' /proc/cpuinfo) ]]; then
```
Comment (imay) [inline build.sh:120]: I have a question. If this is executed on the intel chip and macos, AVX2 and SSE will be disabled.
Could you make them available too?

**Review by decster — APPROVED** [review]

File: build.sh (hunk 3 above)
Comment (decster) [reply build.sh:120]: @luohaha Do you want to improve this on this PR, or I merge this and maybe work on this later?

Comment (luohaha) [reply build.sh:120]: @imay On macos, `/proc` not supported. We can only support compile starrocks on linux now, so I don't think it's a problem.

**Pull request merged**